src/Workspaces/Remote/ServiceHub/Host/ChecksumSynchronizer.cs

```diff
@@ -114,7 +115,25 @@ private void AddIfNeeded(HashSet<Checksum> checksums, IEnumerable<object> checks
                         AddIfNeeded(checksums, checksum);
                         continue;
                     case ChecksumCollection checksumCollection:
-                        AddIfNeeded(checksums, checksumCollection);
+                        AddIfNeeded(checksums, checksumCollection.Children);
```

Comment:
can the containing meethod just be updated to take an ImmutableArray not an IEnumerable?

Reply:
Yep, ported over the changes from your PR to this one in case you would prefer to use this PR for your fix.